src/components/settings/SettingsAppearanceTab/index.jsx

```diff
@@ -1,48 +1,121 @@
 import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { CardGrid, Dropzone, Toast } from '@edx/paragon';
+import {
+  Alert, Button, CardGrid, Dropzone, Image, Toast, useToggle,
+} from '@edx/paragon';
+import { Info } from '@edx/paragon/icons';
 
 import InfoHover from '../../InfoHover';
 import LmsApiService from '../../../data/services/LmsApiService';
 import ThemeCard from './ThemeCard';
+import CustomThemeModal from './CustomThemeModal';
 import {
   ACUMEN_THEME, CAMBRIDGE_THEME, IMPACT_THEME, PIONEER_THEME, SAGE_THEME, SCHOLAR_THEME,
 } from '../data/constants';
 
-export default function SettingsAppearanceTab({
-  enterpriseId,
-}) {
+export const SettingsAppearanceTab = ({
+  enterpriseId, enterpriseBranding, updatePortalConfiguration,
+}) => {
   const logoMessage = 'Your logo will appear on the upper left of every page for both learners and administrators. For best results, use a rectagular logo that is longer in width and has a transparent or white background.';
   const themeMessage = 'Select designer curated theme colors to update the look and feel of your learner and administrator experiences, or create your own theme.';
   const [configChangeSuccess, setConfigChangeSuccess] = useState(null);
-  const [theme, setTheme] = useState(SCHOLAR_THEME);
+  const [uploadedFile, setUploadedFile] = useState(undefined);
+  const [customIsOpen, openCustom, closeCustom] = useToggle(false);
+  const curatedThemes = [ACUMEN_THEME, CAMBRIDGE_THEME, IMPACT_THEME, PIONEER_THEME, SAGE_THEME, SCHOLAR_THEME];
+
+  function getStartingTheme() {
+    for (let i = 0; i < curatedThemes.length; i++) {
+      if (curatedThemes[i].button === enterpriseBranding.primary_color
+        && curatedThemes[i].banner === enterpriseBranding.secondary_color
+        && curatedThemes[i].accent === enterpriseBranding.tertiary_color) {
+        return [curatedThemes[i], null];
+      }
+    }
+    const CUSTOM_THEME = {
+      title: 'Custom Theme',
+      button: enterpriseBranding.primary_color,
+      banner: enterpriseBranding.secondary_color,
+      accent: enterpriseBranding.tertiary_color,
+    };
+    return [CUSTOM_THEME, CUSTOM_THEME];
+  }
 
-  async function handleProcessUpload({
+  // this variable is [selected , null] if the user has not created a custom
+  // theme and it is [selectedTheme, [customName, #primary, #secondary, #tertiary]] if they have
+  // (they can't be two variables because of rules regarding setting react hooks)
```

Comment:
Ah, I see these comments explaining the behavior elsewhere now. Anyway to make the return data types consistent here?